### PR TITLE
8244711: [lworld] Javac needs to tweak checks for interface implementation by an inline type.

### PR DESCRIPTION
**Pull request by sadayapalam**
When javac needs to check if a class implements a particular interface, it should
check if the reference projection of the class is a subtype of the interface. (Rather
than checking if the concerned class is a subtype of the interface)

---------
### Progress
- [x] Change must not contain extraneous whitespace

### Issue
 * [JDK-8244711](https://bugs.openjdk.java.net/browse/JDK-8244711): [lworld] Javac needs to tweak checks for interface implementation by an inline type.

### Reviewers
 * JimLaskey (no known github.com user name / role)

### Download
`$ git fetch https://git.openjdk.java.net/valhalla pull/41/head:pull/41`
`$ git checkout pull/41`

### CODE REVIEW

File: test/langtools/tools/javac/valhalla/lworld-values/ConflictingSuperInterfaceTest.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/*
```
Comment (JimLaskey) [inline test/langtools/tools/javac/valhalla/lworld-values/ConflictingSuperInterfaceTest.java:1]: Missing Copyright.

**Comment by sadayapalam** [comment]
> Since asSuper is always called with a true checkReferenceProjection, why not drop the argument and rename as asSuperWithProjection (or some such). If you introduce a need later for the false case then you can just call asSuper directly. I.E., argument seems redundant.

I agree this looks strange. But rather than asSuperWithProjection, I have reworked it along the lines of existing variants (isSubtype and friends) by making the original asSuper to be a three argument one and all existing calls to the two args asSuper to be routed to the three arg full bodied asSuper with a 3rd argument value of false.

File: test/langtools/tools/javac/valhalla/lworld-values/AutoCloseableTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        String found = "";
+        try (Foo foo = new Foo()) {
+            expected = foo.s;
+        } catch (Exception e) {
+            found = e.getMessage();
+        }
+        if (expected.length() == 0 || !expected.equals(found))
+            throw new AssertionError("Should not get here!");
+    }
+}
```
[the remark this answers is not in the document]
Comment (sadayapalam) [reply test/langtools/tools/javac/valhalla/lworld-values/AutoCloseableTest.java:57]: Will fix. Thanks

File: test/langtools/tools/javac/valhalla/lworld-values/ConflictingSuperInterfaceTest.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+ */
+
+public class ConflictingSuperInterfaceTest {
+
+    interface I<T> {}
+    abstract class S implements I<String> {}
+    inline static class Foo extends S implements I<Integer> {
+        String s = "";
+    }
+}
```
[the remark this answers is not in the document]
Comment (sadayapalam) [reply test/langtools/tools/javac/valhalla/lworld-values/ConflictingSuperInterfaceTest.java:15]: Will fix, thanks!

File: test/langtools/tools/javac/valhalla/lworld-values/ConflictingSuperInterfaceTest.java (hunk 1 above)
Comment (sadayapalam) [reply test/langtools/tools/javac/valhalla/lworld-values/ConflictingSuperInterfaceTest.java:1]: This is intentional. In javac's negative tests we don't add a copyright, but a tag that says nodynamiccopyright. 

I have forgotten the reason we do that, but that is the right behavior :)

**Pull request closed without merging**